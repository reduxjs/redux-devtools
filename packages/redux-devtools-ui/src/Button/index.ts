export { default } from './Button.js';
