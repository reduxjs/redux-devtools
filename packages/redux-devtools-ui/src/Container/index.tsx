import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { useTheme, ThemeData } from '../utils/theme';
import { MainContainerWrapper, ContainerWrapper } from './styles';
import { Theme } from '../themes/default';

interface ContainerFromThemeDataProps {
  children?: React.ReactNode;
  themeData: ThemeData;
  className?: string;
}

const ContainerFromThemeData: React.FunctionComponent<
  ContainerFromThemeDataProps
> = ({ themeData, className, children }) => {
  const theme = useTheme(themeData);
  return (
    <ThemeProvider theme={theme}>
      <MainContainerWrapper className={className}>
        {children}
      </MainContainerWrapper>
    </ThemeProvider>
  );
};

interface Props {
  children?: React.ReactNode;
  themeData?: ThemeData;
  theme?: Theme;
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({
  themeData,
  className,
  theme,
  children,
}) => {
  if (!themeData) {
    return (
      <ContainerWrapper className={className} theme={theme}>
        {children}
      </ContainerWrapper>
    );
  }

  return (
    <ContainerFromThemeData themeData={themeData} className={className}>
      {children}
    </ContainerFromThemeData>
  );
};

Container.propTypes = {
  children: PropTypes.node,
  themeData: PropTypes.any,
  theme: PropTypes.any,
  className: PropTypes.string,
};

export default Container;
