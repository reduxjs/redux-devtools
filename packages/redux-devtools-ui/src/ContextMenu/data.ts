export const items = [
  {
    name: 'Menu Item 1',
  },
  {
    name: 'Menu Item 2',
  },
  {
    name: 'Menu Item 3',
  },
];
