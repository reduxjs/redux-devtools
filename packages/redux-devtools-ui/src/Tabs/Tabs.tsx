import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TabsHeader, { ReactButtonElement, Tab } from './TabsHeader';
import { TabsContainer } from './styles/common';

export type Position = 'left' | 'right' | 'center';

export interface TabsProps<P> {
  tabs: Tab<P>[];
  selected?: string;
  main?: boolean;
  onClick: (value: string) => void;
  collapsible?: boolean;
  position: Position;
}

export default class Tabs<P> extends Component<TabsProps<P>> {
  SelectedComponent?: React.ComponentType<P>;
  selector?: () => P;

  onMouseUp: React.MouseEventHandler<HTMLButtonElement> = (e) => {
    e.currentTarget.blur();
  };

  onClick: React.MouseEventHandler<HTMLButtonElement> = (e) => {
    this.props.onClick(e.currentTarget.value);
  };

  renderTabs() {
    const tabs = this.props.tabs;
    const selected = this.props.selected;

    return tabs.map((tab, i) => {
      let isSelected;
      const value = typeof tab.value !== 'undefined' ? tab.value : tab.name;
      if (value === selected) {
        isSelected = true;
        if (tab.component) {
          this.SelectedComponent = tab.component;
          if (tab.selector) this.selector = () => tab.selector!(tab);
        }
      }
      return (
        <button
          key={i}
          value={value}
          data-selected={isSelected}
          onMouseUp={this.onMouseUp}
          onClick={this.onClick}
        >
          {tab.name}
        </button>
      );
    });
  }

  render() {
    const tabsHeader = (
      <TabsHeader
        tabs={this.renderTabs()}
        items={this.props.tabs}
        main={this.props.main}
        collapsible={this.props.collapsible}
        onClick={this.props.onClick}
        selected={this.props.selected}
        position={this.props.position}
      />
    );

    if (!this.SelectedComponent) {
      return (
        <TabsContainer position={this.props.position}>
          {tabsHeader}
        </TabsContainer>
      );
    }

    return (
      <TabsContainer position={this.props.position}>
        {tabsHeader}
        <div>
          <this.SelectedComponent {...(this.selector! && this.selector())} />
        </div>
      </TabsContainer>
    );
  }

  static propTypes = {
    tabs: PropTypes.array.isRequired,
    selected: PropTypes.string,
    main: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    collapsible: PropTypes.bool,
    position: PropTypes.oneOf(['left', 'right', 'center']),
  };

  static defaultProps = { position: 'left' };
}
