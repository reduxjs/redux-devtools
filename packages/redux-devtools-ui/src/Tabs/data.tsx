import React from 'react';

/* eslint-disable react/prop-types */
const Component = ({ selected }: { selected: string }) => (
  <div
    style={{
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      width: '100%',
      height: '100%',
      fontSize: '22px',
    }}
  >
    Selected {selected}
  </div>
);
/* eslint-enable react/prop-types */

const selector = (tab: { name: string; value?: string }) => ({
  selected: tab.name,
});

export const tabs = [
  {
    name: 'Tab1',
    component: Component,
    selector,
  },
  {
    name: 'Tab2',
    component: Component,
    selector,
  },
  {
    name: 'Tab3',
    component: Component,
    selector,
  },
];

export const simple10Tabs: { name: string; value: string }[] = [];
for (let i = 1; i <= 10; i++)
  simple10Tabs.push({ name: `Tab${i}`, value: `${i}` });
