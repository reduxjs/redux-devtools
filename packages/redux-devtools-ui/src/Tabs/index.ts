export { default } from './Tabs';
export { Tab } from './TabsHeader';
