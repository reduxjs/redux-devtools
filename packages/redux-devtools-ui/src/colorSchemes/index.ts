export { default } from './default';
export { default as github } from './github';
export { default as atomOneDark } from './atom-one-dark';
export { default as dracula } from './dracula';
export { default as iRBlack } from './ir-black';
export { default as macintosh } from './macintosh';
export { default as materia } from './materia';
export { default as oceanicNext } from './oceanic-next';
export { default as phD } from './phd';
export { default as pico } from './pico';
export { default as solarFlare } from './solar-flare';
export { default as spacemacs } from './spacemacs';
export { default as unikitty } from './unikitty';
export { default as woodland } from './woodland';
