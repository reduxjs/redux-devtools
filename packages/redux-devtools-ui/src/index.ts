export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as ContextMenu } from './ContextMenu';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs, Tab } from './Tabs';
export { default as SegmentedControl } from './SegmentedControl';
export { default as Notification } from './Notification';
export * from './Toolbar';

import color from './utils/color';
export const effects = { color };
export { default as createStyledComponent } from './utils/createStyledComponent';
export { Theme, ThemeFromProvider, Scheme } from './utils/theme';
