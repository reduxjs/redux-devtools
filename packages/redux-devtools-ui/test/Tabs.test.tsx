import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Tabs } from '../src';
import { tabs, simple10Tabs } from '../src/Tabs/data';

describe('Tabs', function () {
  it('renders correctly', () => {
    const { container } = render(
      <Tabs
        tabs={tabs}
        onClick={() => {
          // noop
        }}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with props', () => {
    const { container } = render(
      <Tabs
        tabs={tabs}
        onClick={() => {
          // noop
        }}
        selected="Tab2"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders tabs without inner components', () => {
    const { container } = render(
      <Tabs
        tabs={simple10Tabs}
        onClick={() => {
          // noop
        }}
        selected="5"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should select tab', async () => {
    const onClick = jest.fn();
    render(<Tabs tabs={tabs} onClick={onClick} />);

    await userEvent.click(screen.getByRole('button', { name: 'Tab1' }));
    expect(onClick).toHaveBeenCalled();
  });
});
