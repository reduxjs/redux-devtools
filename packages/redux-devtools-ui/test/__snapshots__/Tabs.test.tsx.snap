// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly 1`] = `
<div
  class="sc-iqcoie bAoRWG"
>
  <div
    class="sc-kDDrLX hMIvxf"
  >
    <div>
      <button
        value="Tab1"
      >
        Tab1
      </button>
      <button
        value="Tab2"
      >
        Tab2
      </button>
      <button
        value="Tab3"
      >
        Tab3
      </button>
    </div>
  </div>
</div>
`;

exports[`Tabs renders tabs without inner components 1`] = `
<div
  class="sc-iqcoie bAoRWG"
>
  <div
    class="sc-kDDrLX hMIvxf"
  >
    <div>
      <button
        value="1"
      >
        Tab1
      </button>
      <button
        value="2"
      >
        Tab2
      </button>
      <button
        value="3"
      >
        Tab3
      </button>
      <button
        value="4"
      >
        Tab4
      </button>
      <button
        data-selected="true"
        value="5"
      >
        Tab5
      </button>
      <button
        value="6"
      >
        Tab6
      </button>
      <button
        value="7"
      >
        Tab7
      </button>
      <button
        value="8"
      >
        Tab8
      </button>
      <button
        value="9"
      >
        Tab9
      </button>
      <button
        value="10"
      >
        Tab10
      </button>
    </div>
  </div>
</div>
`;

exports[`Tabs renders with props 1`] = `
<div
  class="sc-iqcoie bAoRWG"
>
  <div
    class="sc-kDDrLX hMIvxf"
  >
    <div>
      <button
        value="Tab1"
      >
        Tab1
      </button>
      <button
        data-selected="true"
        value="Tab2"
      >
        Tab2
      </button>
      <button
        value="Tab3"
      >
        Tab3
      </button>
    </div>
  </div>
  <div>
    <div
      style="display: flex; align-items: center; justify-content: center; width: 100%; height: 100%; font-size: 22px;"
    >
      Selected 
      Tab2
    </div>
  </div>
</div>
`;
